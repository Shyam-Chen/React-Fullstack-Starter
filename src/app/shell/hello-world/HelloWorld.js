// @flow

import React from 'react';
import { compose } from 'recompose';

export const HelloWorld = (): React$Element<*> => (
  <div id="hello-world">
    <h1>Hello, World!</h1>
  </div>
);

export default compose()(HelloWorld);
