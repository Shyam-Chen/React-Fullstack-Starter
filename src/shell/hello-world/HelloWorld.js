import React, { useState } from 'react';
import { useSelector } from 'react-redux';

import { dynamic } from '~/core/store';

// import * as actions from './actions';
// import * as selectors from './selectors';
import reducer from './reducer';

const HelloWorld = () => {
  const [world] = useState('World');
  const { hello } = useSelector(state => state.helloWorld);

  return (
    <div id="hello-world">
      <h1>{hello}, {world}!</h1>
    </div>
  );
};

export default dynamic('helloWorld', reducer)(HelloWorld);
