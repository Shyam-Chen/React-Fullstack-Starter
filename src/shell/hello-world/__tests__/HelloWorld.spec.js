import React from 'react';
import { render } from '@testing-library/react';

import HelloWorld from '../HelloWorld';

describe('HelloWorld', () => {
  it('should render an initial component', () => {
    const { container } = render(<HelloWorld />);
    expect(container).toMatchSnapshot();
  });
});
