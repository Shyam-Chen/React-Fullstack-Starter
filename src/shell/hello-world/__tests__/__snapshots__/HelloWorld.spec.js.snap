// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should render an initial component 1`] = `
<div>
  <div
    id="hello-world"
  >
    <h1>
      Hello
      , 
      World
      !
    </h1>
  </div>
</div>
`;
